core/tee/tee_svc.c

```diff
@@ -110,44 +110,109 @@ uint32_t tee_svc_sys_nocall(void)
 
 TEE_Result tee_svc_sys_get_property(uint32_t prop, tee_uaddr_t buf, size_t blen)
 {
-	static const char api_vers[] = "1.0";
-	static const char descr[] = "Version N.N";
+	/* API implementation version */
+	char api_vers[] = CFG_TEE_API_VERSION;
+
+	/* Implementation description (implementation-dependent) */
+	char descr[] = CFG_TEE_IMPL_DESCR;
+
 	/*
-	 * Value 100 means:
-	 * System time based on REE-controlled timers. Can be tampered by the
-	 * REE.  The implementation must still guarantee that the system time
+	 * System time protection level
+	 * 100: System time based on REE-controlled timers (default).
+	 * Can be tampered by the REE
+	 * The implementation must still guarantee that the system time
 	 * is monotonous, i.e., successive calls to TEE_GetSystemTime must
 	 * return increasing values of the system time.
+	 * 1000: System time based on a TEE-controlled secure timer.
+	 * The REE cannot interfere with the system time. It may still
+	 * interfere with the scheduling of TEE tasks, but is not able to
+	 * hide delays from a TA calling TEE_GetSystemTime.
+	 */
+	uint32_t sys_time_prot_lvl = 100;
+
+	/*
+	 * TA persistent time protection level
+	 * 100: Persistent time based on an REE-controlled real-time clock
+	 * and on the TEE Trusted Storage for the storage of origins (default).
+	 * 1000: Persistent time based on a TEE-controlled real-time clock
+	 * and the TEE Trusted Storage.
+	 * The real-time clock MUST be out of reach of software attacks
+	 * from the REE.
+	 */
+	uint32_t ta_time_prot_lvl = 100;
+
+	/* Elliptic Curve Cryptographic support (false by default) */
+	bool crypto_ecc_en;
+
+	/*
+	 * Trusted storage anti rollback protection level
+	 * 0 (or missing): No antirollback protection (default)
+	 * 100: Antirollback enforced at REE level
+	 * 1000: Antirollback TEE-controlled hardware
 	 */
-	static const uint32_t sys_time_prot_lvl = 100;
-	static const uint32_t ta_time_prot_lvl = 100;
+	uint32_t ts_antiroll_prot_lvl;
+
+	/* Trusted OS implementation version */
+	char trustedos_impl_version[] = CFG_TEE_IMPL_VERSION;
+
+	/* Trusted OS implementation version (binary value) */
+	uint32_t trustedos_impl_bin_version; /* zero length value by default */
+
+	/* Trusted OS implementation manufacturer name */
+	char trustedos_manufacturer[] = CFG_TEE_MANUFACTURER;
+
+	/* Trusted firmware version */
+	char fw_impl_version[] = CFG_TEE_FW_IMPL_VERSION;
+
+	/* Trusted firmware version (binary value) */
+	uint32_t fw_impl_bin_version; /* zero length value by default */
+
+	/* Trusted firmware manufacturer name */
+	char fw_manufacturer[] = CFG_TEE_FW_MANUFACTURER;
+
 	struct tee_ta_session *sess;
 	TEE_Result res;
 
+	struct tee_props {
+		void *data;
+		size_t len;
+	};
+
+	/* Consistent with enum utee_property */
+	const struct tee_props tee_props_lut[] = {
```

Comment:
To save stack (and unneeded re-initialization each time this function is entered) I'd like to have this outside of this function. All the config variables would then also need to be moved outside this function, preferably with a const attribute.